base/asyncevent.jl

```diff
@@ -45,13 +45,22 @@ the async condition object itself.
 """
 function AsyncCondition(cb::Function)
     async = AsyncCondition()
-    t = @task while _trywait(async)
-        cb(async)
-        isopen(async) || return
+    t = @task begin
+        unpreserve_handle(async)
```

Comment:
Isn't `async` reachable through `t.code`? Why would it be finalized in the old code?

Reply:
t is only reachable through async

Reply:
ahhhh